test/test_fx_experimental.py

```diff
@@ -162,5 +156,51 @@ def forward(self, a, b, offset):
         self.assertEqual(traced(a, b, offset), module_with_submodules(a, b, offset))
         assert len(module_with_submodules.graph.nodes) == 24
 
-if __name__ == '__main__':
+    def test_serialize_graph(self):
+        class TestModule(torch.nn.Module):
+            def __init__(self):
+                super().__init__()
+                self.linear = torch.nn.Linear(4, 4)
+                self.e = torch.rand(4)
+
+            def forward(self, a, b):
+                add_1 = a + b
+                linear = self.linear(add_1)
+                add_2 = linear + self.e
+                return add_2
+
+        m = TestModule()
+        traced = symbolic_trace(m)
+        a = torch.rand(4)
+        b = torch.rand(4)
+        GraphManipulation.get_size_of_all_nodes(traced, [a, b])
+
+        partitioner = Partitioner()
+        devices = [Device("dev_0", 5000, 0), Device("dev_1", 125, 1)]
+        partitioner_config = PartitionerConfig(devices, is_sparse_nn=True)
+        ret = partitioner.partition_graph(traced, m, partitioner_config)
+        module_with_submodules = ret.module_with_submodules
+        # Fix for now to add type/shape to output
+        for node in traced.graph.nodes:
+            if node.op == "output":
+                node.shape = a.shape
+                node.dtype = a.dtype
+        for mod in module_with_submodules.modules():
+            if isinstance(mod, torch.fx.GraphModule):
+                for node in mod.graph.nodes:
+                    node.shape = a.shape
+                    node.dtype = a.dtype
+        for node in module_with_submodules.graph.nodes:
+            node.shape = a.shape
+            node.dtype = a.dtype
```

Comment:
What's going on with all this shape stuff?

Reply:
poor man's shape propagation. The serialization requires it to be present but for testing serialization it doesn't matter what the info is.